dawn/test/unit-test/dawn/CodeGen/Cuda/TestCodeGenCuda.cpp

```diff
@@ -0,0 +1,76 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/CodeGen/CodeGen.h"
+#include "dawn/CodeGen/Cuda/CudaCodeGen.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/Support/DiagnosticsEngine.h"
+#include "dawn/Unittest/IIRBuilder.h"
+#include "dawn/Unittest/UnittestLogger.h"
+
+#include <gtest/gtest.h>
+
+#include <cstring>
+#include <fstream>
+
+namespace {
+
+void dump(std::ostream& os, dawn::codegen::stencilInstantiationContext& ctx) {
+  dawn::DiagnosticsEngine diagnostics;
+  using CG = dawn::codegen::cuda::CudaCodeGen;
+  CG generator(ctx, diagnostics, 0, 0, 0, {0, 0, 0});
+  auto tu = generator.generateCode();
+
+  std::ostringstream ss;
+  for(auto const& macroDefine : tu->getPPDefines())
+    ss << macroDefine << "\n";
+
+  ss << tu->getGlobals();
+  for(auto const& s : tu->getStencils())
+    ss << s.second;
+  os << ss.str();
+}
+
+std::string read(const std::string& file) {
```

Comment:
should we export this to `dawn/dawn/src/unittest/` or `dawn/dawn/src/support/` as support functions that are used in unittesting?